sycl/include/CL/sycl/buffer.hpp

```diff
@@ -21,7 +21,7 @@ class queue;
 template <int dimensions> class range;
 
 template <typename T, int dimensions = 1,
-          typename AllocatorT = cl::sycl::buffer_allocator>
+          typename AllocatorT = cl::sycl::detail::aligned_allocator<T>>
```

Comment:
align with specification

Reply:
> align with specification

Spec is incorrect at this point. buffer_allocator must be a template class for various of reasons, but it's not. The issue will be created, but for now I would stay with the approach made by Sergey.

Reply:
So, what I mean, I would approve the change unless Alexey hasn't strong objections of doing so.

Reply:
No objections from me.